core/store/src/flat/delta.rs

```diff
@@ -47,6 +54,16 @@ where
     }
 }
 
+impl std::fmt::Display for FlatStateChanges {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> Result<(), std::fmt::Error> {
+        if self.len() <= 5 {
+            write!(f, "{:?}", self.0)
+        } else {
+            write!(f, "{} items", self.len())
+        }
```

Comment:
As @pugachAG hinted, this isn't very typical for `Display`. Usually you wouldn't want to remove significant information from a `Display` implementation. It also spoils `to_string()` in perhaps unexpected ways.

I would prefer one of two options.
1) Either change it to `fmt::Debug` (here and for `FlatStateDelta`) if you think it's generally useful anytime you want to debug runtime data through logs.
2) Or else, if you want to keep the default `Debug` format, then create a wrapper type and implement `fmt::Display` for that. This is also described as "adapter" in [RFC 565](https://github.com/rust-lang/rfcs/blob/master/text/0565-show-string-guidelines.md#user-facing-fmtdisplay). Following that example closely, you could create `struct ForMyFancyFlatStateCommand<'a, T>(&'a T);` and implement `impl<'a> fmt::Display for ForMyFancyFlatStateCommand<'a, FlatStateChanges> { ... }` plus `impl<'a> fmt::Display for ForMyFancyFlatStateCommand<'a, FlatStateDelta> { ... }` or something along those line.